template.tpl

```diff
@@ -1074,6 +1077,178 @@ scenarios:
     // Call runCode to run the template's code.
     let variableResult = runCode(mockData);
     assertThat(variableResult).isEqualTo(expected);
+- name: Test anonymous tracking true - options true
+  code: |
+    const mockData = {
+      trackerName: 'spTracker',
+      collectorEndpoint: 'test',
+
+      spLibrary: 'jsDelivr',
+      version: '3.5.0',
+
+      appId: 'my-site',
+      platform: 'web',
+
+      respectDoNotTrack: true,
+      anonymousTracking: 'anonymousTrackingTrue',
+      withSessionTracking: true,
+      withServerAnonymisation: true,
+
+      stateStorageStrategy: 'cookieAndLocalStorage',
+      cookieDomain: 'auto',
+      cookieName: 'sp',
+      cookieLifetime: 63072000,
+      cookieSameSite: 'Lax',
+      sessionCookieTimeout: '1800',
+      cookieSecure: true,
+      maxLocalStorageQueueSize: '1000',
+
+      eventMethod: 'post',
+      postPath: '/com.snowplowanalytics.snowplow/tp2',
+      bufferSize: '1',
+      encodeBase64: true,
+      maxPostBytes: '40000',
+      connectionTimeout: '5000',
+
+      webPage: true,
+      gaCookies: false,
+      clientHints: false,
+      performanceTiming: false,
+      geolocation: false,
+    };
```

Comment:
This looks like it will get pretty repetitious, which is a concern if the tests get flakey as the line count goes up as you mentioned. Should we use the `setup` code to have like a base object that we can just have overrides for on specific properties?

There's enough in the std lib for a pseudo-`Object.assign()` with something like:

```javascript
const extendMock = (overrides) => {
   const Object = require('Object');
   const baseMockData = {
     // base template properties here
   };

  return Object.entries(overrides || {}).reduce((result, entry) => {
    result[entry[0]] = entry[1];
    return result;
  }, baseMockData);
};
```

Then used like:

```javascript
const mockData = extendMock({
  trackerName: 'spTracker',
  collectorEndpoint: 'test',
});

let variableResult = runCode(mockData);
```

Reply:
This is a great idea and definitely one we can consider for other gtm templates as well! I opened [this issue](https://github.com/snowplow/snowplow-gtm-variable-template-v3/issues/22) for reference.